cluster-autoscaler/expander/factory/expander_factory.go

```diff
@@ -17,24 +17,35 @@ limitations under the License.
 package factory
 
 import (
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
 	"k8s.io/autoscaler/cluster-autoscaler/expander"
 	"k8s.io/autoscaler/cluster-autoscaler/expander/mostpods"
+	"k8s.io/autoscaler/cluster-autoscaler/expander/price"
 	"k8s.io/autoscaler/cluster-autoscaler/expander/random"
 	"k8s.io/autoscaler/cluster-autoscaler/expander/waste"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+
+	kube_util "k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
 )
 
 // ExpanderStrategyFromString creates an expander.Strategy according to its name
-func ExpanderStrategyFromString(expanderFlag string) expander.Strategy {
-	var expanderStrategy expander.Strategy
-	{
-		switch expanderFlag {
-		case expander.RandomExpanderName:
-			expanderStrategy = random.NewStrategy()
-		case expander.MostPodsExpanderName:
-			expanderStrategy = mostpods.NewStrategy()
-		case expander.LeastWasteExpanderName:
-			expanderStrategy = waste.NewStrategy()
+func ExpanderStrategyFromString(expanderFlag string, cloudProvider cloudprovider.CloudProvider,
+	nodeLister kube_util.NodeLister) (expander.Strategy, error) {
```

Comment:
Just for reference - I'm ok with both returning error here and repacking it to AutoscalerError later and returning AutoscalerError from here.